poorwsgi/request.py

```diff
@@ -891,16 +899,22 @@ def getfirst(self, key=None, default=None, fce=str):
         """
         return fce(self.getvalue(default=default))
 
-    def getlist(self, key=None, fce=str):
+    def getlist(self, key=None, default=None, fce=str):
         """Returns generator of values.
 
         key : None
             Compatibility parametr is ignored.
         fce : convertor (str)
             Function which processed value.
+        default : None
+            Default value when self is empty.
         """
-        for it in self:
-            yield fce(it)
+        if not self:
+            for it in (default or []):
```

Comment:
why not set default's default value in signature as `[]`? It will indicate, that it has to be an iterable and then you can do `for it in default` here directly.